app/main/controller/presto_controller.py

```diff
@@ -0,0 +1,23 @@
+from flask import request, current_app as app
+from flask_restplus import Resource, Namespace
+
+from app.main.lib.fields import JsonObject
+from app.main.lib import similarity
+
+api = Namespace('presto', description='presto callback url')
+presto = api.model('presto', {
+    'body': JsonObject(required=False, description='Original body of the message sent - will contain ID, callback_url, and original url/text of request'),
+    'response': JsonObject(required=False, description='The response data sent back from the model'),
+})
+@api.route('/receive/<string:action>/<string:model_type>')
+class PrestoResource(Resource):
+    @api.response(200, 'Successfully responded to presto callback.')
+    @api.doc('Receive a presto callback for a given `model_type`')
+    def post(self, action, model_type):
+        data = request.args or request.json
+        if action == "add_item":
```

Comment:
lets add delete also, if we if throw 'not implement' for now

Reply:
I don't know if this is the right direction to take this code. If I were to implement this right now, I'd want to define a `similarity.callback_delete_item` method, which just goes right back to the delete command that would (and should) otherwise use the existing functionality (which basically is just adding another set of unnecessary code right now). If I don't do that, and raise a not implemented error, I'd just end up with functionally identical code:

```
    def post(self, action, model_type):
        data = request.args or request.json
        if action == "add_item":
            return similarity.callback_add_item(data, model_type)
        elif action == "search_item":
            return similarity.callback_get_similar_items(data, model_type)
        elif action == "delete_item":
            abort(404, description=f"Action type of {action} was not found. Currently available action types are add_item, search_item.")
        else:
            abort(404, description=f"Action type of {action} was not found. Currently available action types are add_item, search_item.")
```
VS:

```
    def post(self, action, model_type):
        data = request.args or request.json
        if action == "add_item":
            return similarity.callback_add_item(data, model_type)
        elif action == "search_item":
            return similarity.callback_get_similar_items(data, model_type)
        else:
            abort(404, description=f"Action type of {action} was not found. Currently available action types are add_item, search_item.")
```

I think this may be some overengineering at this point of development. If it's ok with you, I would really prefer we hold on designing more in this direction for now?

Reply:
agreed that the generic `else` clause can capture this.